src/backend/distributed/master/master_delete_protocol.c

```diff
@@ -319,14 +319,21 @@ DropShards(Oid relationId, char *schemaName, char *relationName,
 	ListCell *shardIntervalCell = NULL;
 	int droppedShardCount = 0;
 
+	if (XactModificationLevel != XACT_MODIFICATION_NONE)
+	{
+		ereport(ERROR, (errcode(ERRCODE_ACTIVE_SQL_TRANSACTION),
+						errmsg("shard drop operations must not appear in "
+							   "transaction blocks containing other distributed "
+							   "modifications")));
+	}
+
+	BeginOrContinueCoordinatedTransaction();
+	CoordinatedTransactionUse2PC();
```

Comment:
I add it to not DROP shards before being sure there will not be any errors in other shards. However since we use coordinated transaction, it does not necessary in this case any more. I changed it to be dependent on the multi_shard_commit_protocol GUC.